tests/signal/tst_signal.cpp

```diff
@@ -77,6 +79,128 @@ TEST_CASE("Signal connections")
         REQUIRE(lambdaCalled == true);
     }
 
+    SUBCASE("Disconnect Deferred Connection")
+    {
+        Signal<int> signal;
+        int val = 4;
+        auto evaluator = std::make_shared<ConnectionEvaluator>();
+
+        auto connection = signal.connectDeferred(evaluator, [&val](int value) {
+            val += value;
+        });
+
+        REQUIRE(connection.isActive());
+
+        signal.emit(4);
+
+        connection.disconnect();
+        REQUIRE(!connection.isActive());
+
+        signal.emit(6); // It will not affect the result as the signal is disconnected
+
+        evaluator->evaluateDeferredConnections();
+        REQUIRE(val == 8);
+    }
+
+    SUBCASE("Multiple Signals with Evaluator")
+    {
+        Signal<int> signal1;
+        Signal<int> signal2;
+        int val = 4;
+        auto evaluator = std::make_shared<ConnectionEvaluator>();
+
+        std::thread thread1([&] {
+            signal1.connectDeferred(evaluator, [&val](int value) {
+                val += value;
+            });
+        });
+
+        std::thread thread2([&] {
+            signal2.connectDeferred(evaluator, [&val](int value) {
+                val += value;
+            });
+        });
+
+        thread1.join();
+        thread2.join();
+
+        signal1.emit(2);
+        signal2.emit(3);
+
+        evaluator->evaluateDeferredConnections();
+
+        REQUIRE(val == 9);
+    }
+
+    SUBCASE("Emit Multiple Signals with Evaluator")
+    {
+        Signal<int> signal1;
+        Signal<int> signal2;
+        int val1 = 4;
+        int val2 = 4;
+        auto evaluator = std::make_shared<ConnectionEvaluator>();
+
+        signal1.connectDeferred(evaluator, [&val1](int value) {
+            val1 += value;
+        });
+
+        signal2.connectDeferred(evaluator, [&val2](int value) {
+            val2 += value;
+        });
+
+        std::thread thread1([&] {
+            signal1.emit(2);
+        });
+
+        std::thread thread2([&] {
+            signal2.emit(3);
+        });
+
+        thread1.join();
+        thread2.join();
+
```

Comment:
```suggestion
REQUIRE(val1 == 4);
REQUIRE(val2 == 4);
```

To make sure the values don't immediately change, when emitting, assert that they're still the initial value right before calling `evaluate`.

Reply:
Also applies to the other tests